addon/components/nrg-dropdown/component.js

```diff
@@ -79,6 +82,37 @@ export default Component.extend(Validation, EKMixin, EKFirstResponderOnFocusMixi
     document.removeEventListener('click', this._clickHandler, true);
   },
 
+  addMouseListeners() {
+    document.addEventListener('mouseenter', this._mouseEnterHandler, true);
+    document.addEventListener('mouseleave', this._mouseLeaveHandler, true);
+  },
+
+  removeMouseListeners() {
+    document.removeEventListener('mouseenter', this._mouseEnterHandler, true);
+    document.removeEventListener('mouseleave', this._mouseLeaveHandler, true);
+  },
+
+  createMouseEnterHandler() {
+    this.set('_mouseEnterHandler', evt => {
+      if (this.element && this.element.contains(evt.target)) {
+        const item = evt.target.closest('[data-dropdown-item]');
+        if(item){
+          const parent = item.parentNode;
+          const index = [].indexOf.call(parent.children, item);
```

Comment:
You could change `data-dropdown-item` to be the item index instead of just being `true`.
That way you can get the index instantly.
Or just create another `data` attribute for it. it feels a little saver.
This might be something that can help with the scroll fix as well!